src/cmap/manager.rs

```diff
@@ -20,8 +20,15 @@ pub(super) struct PoolManager {
 
 impl PoolManager {
     /// Lazily clear the pool.
-    pub(super) fn clear(&self) {
-        let _ = self.sender.send(PoolManagementRequest::Clear);
+    pub(super) async fn clear(&self) {
```

Comment:
Because clearing the pool needs to occur only while the lock on the topology is held, this was turned into a blocking operation.